test/com/amazon/ion/impl/bin/IonManagedBinaryWriterTest.java

```diff
@@ -104,6 +147,21 @@ public void testFlushImmediatelyAfterIVM() throws Exception
         assertEquals("burrito", ((IonSymbol) dg.systemGet(2)).stringValue());
     }
 
+    @Test
+    public void testAutoFlush() throws Exception{
+
+        IonReader reader = system().newReader(source.toByteArray());
+        ByteArrayOutputStream actual = new ByteArrayOutputStream();
+        IonWriter actualWriter = IonBinaryWriterBuilder.standard().withAutoFlushEnbaled(true).build(actual);
+        while (reader.next() != null) {
+            actualWriter.writeValue(reader);
+        }
+        actualWriter.finish();
+        if (lstAppendMode.isEnabled() && autoFlushMode.isEnabled()) {
+            assertArrayEquals(actual.toByteArray(), expectedOut.toByteArray());
+        }
```

Comment:
Let's add an assertion after this branch to make sure that in all cases `actual` and `expectedOut` are data model equivalent.